heat/core/io.py

```diff
@@ -895,8 +895,14 @@ def load_csv(
         # In case there are some empty lines in the csv file
         local_tensor = local_tensor[:actual_length]
 
-        resulting_tensor = factories.array(
-            local_tensor, dtype=dtype, is_split=0, device=device, comm=comm
+        resulting_tensor = DNDarray(
+            local_tensor,
+            gshape=tuple(local_tensor.shape),
```

Comment:
`gshape` is supposed to be the global shape of the memory-distributed array `resulting_tensor`. Here you're setting it to the shape of `local_tensor`, which is basically a slice of the global array.

If we have all the information we need to calculate `gshape` without communication among processes, then we can call `DNDarray(...)`, otherwise we need to use `factories.array()` and that will take care of the comm

Reply:
Okay, understood. So can I make a copy of that tensor before slicing it and pass `gshape=tuple(local_tensor_copy.shape)` to it?